server/server.go

```diff
@@ -454,6 +455,55 @@ func New(
 	return s, nil
 }
 
+// getHostIP first attempts to read the IP address of the host from /proc/net/route
+// If that fails (there is no /proc/net/route), it then attempts to deterministically find a default IP
+// It does so by finding exactly one IP address that is not a loop back, and not created by CNI.
+func (s *Server) getHostIP() (net.IP, error) {
+	// First, attempt to find a primary IP from /proc/net/route deterministically
+	hostIP, err := knet.ChooseBindAddress(nil)
+	if err != nil {
+		// if that fails, check if we can find a primary IP address unambiguously
+		ifaces, err2 := net.Interfaces()
+		if err2 != nil {
+			return nil, errors.Wrapf(err, "Failed to read network Interfaces after failing to choose bind address")
+		}
+
+		// we are hoping for exactly 1 possible IP
+		numPossibleIPs := 0
+		for _, iface := range ifaces {
```

Comment:
could we break this loop into a different function that we can unit test?

Reply:
gah, seems like it make a ton of different calls though that we'd have to mock up :disappointed: